src/sentry/api/method/definition.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+
+def sentry_api():
+    """
+    This decorator handles the API overheads like validating and deserializing inputs,
+    serializing outputs, paginating lists, and more feature to be added in the future.
+
+    It works with the SentrySchema to document everything around the endpoint.
+
+    ```
+    class ApiEndpoint(Endpoint):
+
+        @sentry_api(
+            public=True,
+            query_params=[Serializer1, Serializer2],
```

Comment:
Does this require that we use a DRF serializer for input?

Reply:
In some places we manually instantiate DRF serializers to pass them extra context. How do we handle that case here?

Eg https://github.com/getsentry/sentry/blob/daa39765c1f8b0431cb5bdce7de67b05bec75b1e/src/sentry/incidents/endpoints/project_alert_rule_details.py#L33-L42

Reply:
Hmm, I have to think of a way to do that in the wrapper, but for now, it shouldn't stop the old way from working. I'd just have to add a flag to not validate and only use the serializer for documentation.